.pre-commit-config.yaml

```diff
@@ -1,24 +1,37 @@
 repos:
-  - repo: https://github.com/Quantco/pre-commit-mirrors-ruff
-    rev: 0.4.7
+  - repo: local
     hooks:
-      - id: ruff-conda
-        exclude: ^src/glum_benchmarks/orig_sklearn_fork/
-      - id: ruff-format-conda
-        exclude: ^src/glum_benchmarks/orig_sklearn_fork/
-  - repo: https://github.com/Quantco/pre-commit-mirrors-mypy
-    rev: "1.10.0"
-    hooks:
-      - id: mypy-conda
-        args:
-          - --check-untyped-defs
-          - --ignore-missing-imports
-          - --namespace-packages
+      # ruff
+      - id: ruff
+        name: ruff
+        entry: pixi run -e lint ruff check --fix --exit-non-zero-on-fix --force-exclude
+        language: system
+        types_or: [python, pyi]
+        require_serial: true
+      - id: ruff-format
+        name: ruff-format
+        entry: pixi run -e lint ruff format --force-exclude
+        language: system
+        types_or: [python, pyi]
+        require_serial: true
+      # cython-lint
+      - id: cython-lint
+        name: cython-lint
+        entry: pixi run -e lint cython-lint --no-pycodestyle
+        language: system
+        types: [cython]
+        require_serial: true
+      - id: cython-double-quote
+        name: cython-double-quote
+        entry: pixi run -e lint double-quote-cython-strings
+        language: system
+        types: [cython]
+        require_serial: true
+      # mypy
+      - id: mypy
+        name: mypy
+        entry: pixi run -e default mypy --check-untyped-defs --ignore-missing-imports --namespace-packages
```

Comment:
```suggestion
        entry: pixi run -e default mypy
```

I would suggest moving `--check-untyped-defs` and `--namespace-packages` to the `pyproject.toml` confiiguration and removing `--ignore-missing-imports` altogether as it can lead to issues.

Instead, I would suggest installing additional types if needed (like `pandas-stubs`, `types-psutil`, ...) to your `test` feature next to `mypy` and for dependencies that don't have stubs yet adding an exception to `pyproject.toml` as follows:

```toml
[[tool.mypy.overrides]]
# https://github.com/scikit-learn/scikit-learn/issues/16705
module = ["sklearn.*"]
ignore_missing_imports = true
```